jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/BuildFileSpec.java

```diff
@@ -66,12 +68,14 @@
   @Nullable private BaseImageSpec from;
   @Nullable private Instant creationTime;
   @Nullable private ImageFormat format;
-  @Nullable private Map<String, String> environment;
-  @Nullable private Map<String, String> labels;
-  @Nullable private Set<AbsoluteUnixPath> volumes;
-  @Nullable private Set<Port> exposedPorts;
+  private Map<String, String> environment;
+  private Map<String, String> labels;
+  private Set<AbsoluteUnixPath> volumes;
+  private Set<Port> exposedPorts;
   @Nullable private String user;
   @Nullable private AbsoluteUnixPath workingDirectory;
+  // these two specific collection can be null as they can be set to empty to override
+  // the base image
```

Comment:
This confuses me. It says "can be null" and then "set to empty". Does "empty" mean an empty list? And what exactly "override" means?